### PR TITLE
[bitnami/memcached] adding HPA

### PR DESCRIPTION
**Pull request by cabrinha**
**Description of the change**

Adding HPA to memcached

**Benefits**

Autoscaling

**Possible drawbacks**

... It's disabled by default

**Applicable issues**

  - fixes https://github.com/bitnami/charts/issues/7216

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/memcached/values.yaml [hunk 1]
```diff
@@ -136,6 +136,21 @@ service:
   ## @param service.annotations Additional annotations for Memcached service
   ##
   annotations: {}
+
+## Memcached Autoscaling
+## @param autoscaling.enabled Enable memcached statefulset autoscaling
```
Comment (juan131) [inline bitnami/memcached/values.yaml:141]: This feature only works for HA architecture. Let's state that:
```suggestion
## @param autoscaling.enabled Enable memcached statefulset autoscaling (requires architecture: "high-availability")
```

File: bitnami/memcached/templates/hpa.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.autoscaling.enabled }}
```
Comment (juan131) [inline bitnami/memcached/templates/hpa.yaml:1]: Only for HA architecture:
```suggestion
{{- if and .Values.Values.autoscaling.enabled (eq .Values.architecture "high-availability") }}
```

File: bitnami/memcached/templates/hpa.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        name: cpu
+        targetAverageUtilization: {{ .Values.autoscaling.targetCPU }}
+    {{- end }}
+    {{- if .Values.autoscaling.targetMemory }}
+    - type: Resource
+      resource:
+        name: memory
+        targetAverageUtilization: {{ .Values.autoscaling.targetMemory  }}
+    {{- end }}
+{{- end }}
```
Comment (juan131) [inline bitnami/memcached/templates/hpa.yaml:34]: Add missing new line at the end:
```suggestion
{{- end }}

```

File: bitnami/memcached/templates/hpa.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "memcached.fullname" . }}
```
Comment (juan131) [inline bitnami/memcached/templates/hpa.yaml:5]: Use this macro instead:
```suggestion
  name: {{ template "common.names.fullname" . }}
```

File: bitnami/memcached/templates/hpa.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  scaleTargetRef:
+    apiVersion: {{ include "common.capabilities.statefulset.apiVersion" . }}
+    kind: StatefulSet
+    name: {{ include "memcached.fullname" . }}
```
Comment (juan131) [inline bitnami/memcached/templates/hpa.yaml:18]: Same here:
```suggestion
    name: {{ template "common.names.fullname" . }}
```

File: bitnami/memcached/Chart.yaml [hunk 6]
```diff
@@ -22,4 +22,4 @@ name: memcached
 sources:
   - https://github.com/bitnami/bitnami-docker-memcached
   - http://memcached.org/
-version: 5.13.7
+version: 5.13.8
```
Comment (juan131) [inline bitnami/memcached/Chart.yaml:25]: We're adding a new feature, therefore we should bump the minor version:
```suggestion
version: 5.14.0
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for your contribution @cabrinha !!!!
Please take a look to the comments I added when you have a chance.

**Comment by cabrinha** [comment]
> Thanks so much for your contribution @cabrinha !!!!
> Please take a look to the comments I added when you have a chance.

Done!

File: bitnami/memcached/templates/hpa.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{{- if and .Values.Values.autoscaling.enabled (eq .Values.architecture "high-availability") }}
```
Comment (rafariossaa) [inline bitnami/memcached/templates/hpa.yaml:1]: I think there is a typo here: `.Values.Values.autoscaling.enabled`

**Review by rafariossaa — CHANGES_REQUESTED** [review]

File: bitnami/memcached/templates/hpa.yaml (hunk 7 above)
Comment (cabrinha) [reply bitnami/memcached/templates/hpa.yaml]: Yeah, leftovers from find/replace.

**Review by rafariossaa — APPROVED** [review]
LGTM

**Pull request merged**